src/torchcodec/_core/SingleStreamDecoder.cpp

```diff
@@ -776,7 +785,8 @@ FrameBatchOutput SingleStreamDecoder::getFramesPlayedAt(
     frameIndices[i] = secondsToIndexLowerBound(frameSeconds);
   }
 
-  return getFramesAtIndices(frameIndices);
+  // TODO: Support tensors natively instead of a vector to avoid a copy.
```

Comment:
@mollyxu, can you create an issue to track this? Specifically, we want to apply the same kind of refactoring to `get_frames_displayed_at()`.